### PR TITLE
Added ServiceInfo web API servlet

### PR DESCRIPTION
**Pull request by rftemple**
Adding, updating and removing owner svi via http api's

### CODE REVIEW

File: protocol-samples/http-client-to2-sample/src/main/java/org/fido/iot/sample/To2ClientApp.java [hunk 1]
```diff
@@ -74,6 +74,8 @@ private void run(String[] args)
 
     DispatchResult dr = clientService.getHelloMessage();
 
+    clientService.setTo1d(to1dBlob);
```
Comment (trbehera) [inline protocol-samples/http-client-to2-sample/src/main/java/org/fido/iot/sample/To2ClientApp.java:77]: Is this change required?

Comment (rftemple) [reply protocol-samples/http-client-to2-sample/src/main/java/org/fido/iot/sample/To2ClientApp.java:77]: yes that is needed because to1dBlob verification was added

File: util/storage-samples/storage-owner-sample/src/main/java/org/fido/iot/storage/OwnerSysModule.java [hunk 2]
```diff
@@ -142,10 +142,17 @@ private Composite getNextMessage(Composite resource) {
 
     message.set(Const.FIRST_KEY, contentType);
     switch (contentType) {
-      case FdoSys.KEY_ACTIVE:
-        message.set(Const.SECOND_KEY, true);
+      case FdoSys.KEY_ACTIVE: {
+
+        byte[] content = new OwnerDbManager().getSystemResourceContent(dataSource, resId);
+        if (content.length == 1 && content[0] == -11) {
```
Comment (prateekcr) [inline util/storage-samples/storage-owner-sample/src/main/java/org/fido/iot/storage/OwnerSysModule.java:148]: What is -11 here? Shouldn't it be F4 or F5 as a CBOR byte?

Comment (trbehera) [reply util/storage-samples/storage-owner-sample/src/main/java/org/fido/iot/storage/OwnerSysModule.java:148]: Hex value F5 is -11 when read as signed character.

File: http-api-samples/http-api-owner-sample/src/main/java/org/fido/iot/api/OwnerSystemResourceServlet.java [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+    String varName = req.getParameter(VAR_TAG);
+    String contentType = null;
+
+    if (module != null && varName != null) {
+      contentType = module + ModuleManager.MODULE_DELIMITER + varName;
+    }
+
+    StringBuilder whereString = new StringBuilder("WHERE ");
+
+    if (id != null) {
```
Comment (johneasterday) [inline http-api-samples/http-api-owner-sample/src/main/java/org/fido/iot/api/OwnerSystemResourceServlet.java:76]: Better practice would be to build a parameterized where clause (using ?) instead of adding each as a string to prevent sql injection (this would likely be identified in a security review).

File: http-api-samples/http-api-owner-sample/src/main/java/org/fido/iot/api/OwnerSystemResourceServlet.java [hunk 4]
```diff
@@ -0,0 +1,457 @@
...
+      whereString.append("RESOURCE_ID = ");
+      whereString.append(id.toString());
+    } else {
+      String and = "";
+      if (crid != null) {
+        whereString.append(and);
+        whereString.append("CONTENT_RESOURCE_TAG = ");
+        whereString.append(crid.toString());
+        and = WHERE_AND;
```
Comment (johneasterday) [inline http-api-samples/http-api-owner-sample/src/main/java/org/fido/iot/api/OwnerSystemResourceServlet.java:85]: An easier why to handle the "AND" clause is to structure the  where clause to start with "where 1=1". Then each subsequent optional item can always include the "and". For example: "and some_id = value"

File: http-api-samples/http-api-owner-sample/src/main/java/org/fido/iot/api/OwnerSystemResourceServlet.java [hunk 5]
```diff
@@ -0,0 +1,457 @@
...
+    } catch (SQLException e) {
+      resp.setStatus(Const.HTTP_INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  protected void doPut(HttpServletRequest req, HttpServletResponse resp)
+      throws ServletException, IOException {
+
+    if (req.getContentType().compareToIgnoreCase("application/octet-stream") != 0) {
+      resp.setStatus(415); //unsupported meida
```
Comment (johneasterday) [inline http-api-samples/http-api-owner-sample/src/main/java/org/fido/iot/api/OwnerSystemResourceServlet.java:294]: Define status values as constants?

File: http-api-samples/http-api-owner-sample/src/main/java/org/fido/iot/api/OwnerSystemResourceServlet.java [hunk 6]
```diff
@@ -0,0 +1,457 @@
...
+    String module = req.getParameter(MODULE_TAG);
+    String varName = req.getParameter(VAR_TAG);
+    String contentType = null;
+
+    if (module != null && varName != null) {
+      contentType = module + ModuleManager.MODULE_DELIMITER + varName;
+    }
+
+    if (contentType == null) {
+      resp.setStatus(400); //bad request
```
Comment (johneasterday) [inline http-api-samples/http-api-owner-sample/src/main/java/org/fido/iot/api/OwnerSystemResourceServlet.java:317]: Define status constants?

File: util/storage-samples/storage-owner-sample/src/main/java/org/fido/iot/storage/OwnerSysModule.java (hunk 2 above)
Comment (johneasterday) [reply util/storage-samples/storage-owner-sample/src/main/java/org/fido/iot/storage/OwnerSysModule.java:148]: We should add a comment. Also, could we use 0xF5 instead of -11.

File: http-api-samples/http-api-owner-sample/src/main/java/org/fido/iot/api/OwnerSystemResourceServlet.java (hunk 4 above)
Comment (rftemple) [reply http-api-samples/http-api-owner-sample/src/main/java/org/fido/iot/api/OwnerSystemResourceServlet.java:85]: The reason I did it this way is that I'm not certain  which column will be first.  The first non-null query argument will be the first  in the where clause but any of the columns can to the first.  Whatever strategy I use I will need to keep track using some flag for state on whether the its the first column or non-first column.

Comment (johneasterday) [reply http-api-samples/http-api-owner-sample/src/main/java/org/fido/iot/api/OwnerSystemResourceServlet.java:85]: That is why the where clause always starts with "where 1=1". Then each subsequent clause will always be of the form "and foo=bar". In this case: "and ZZZ_TAG = value".

**Review by johneasterday — APPROVED** [review]

**Review by trbehera — APPROVED** [review]
